Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/Middlewares/AWSS3ErrorWith200StatusXMLMiddleware.swift

```diff
@@ -5,30 +5,25 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+import enum Smithy.ByteStream
 import class Smithy.Context
 import ClientRuntime
 import SmithyHTTPAPI
+import SmithyXML
+import struct Foundation.Data
+import SmithyStreams
 
 public struct AWSS3ErrorWith200StatusXMLMiddleware<OperationStackInput, OperationStackOutput> {
     public let id: String = "AWSS3ErrorWith200StatusXMLMiddleware"
     private let errorStatusCode: HTTPStatusCode = .internalServerError
 
     public init() {}
 
-    private func isErrorWith200Status(response: HTTPResponse) async throws -> Bool {
-        // Check if the status code is OK (200)
-        guard response.statusCode == .ok else {
-            return false
-        }
-
-        // Check if the response body contains an XML Error
-        guard let data = try await response.body.readData() else {
-            return false
-        }
+    private func isRootErrorElement(data: Data) throws -> Bool {
+        let reader = try Reader.from(data: data)
 
-        response.body = .data(data)
-        let xmlString = String(decoding: data, as: UTF8.self)
-        return xmlString.contains("<Error>")
+        // Check if there's an "Error" node at the root of the XML response
+        return reader.nodeInfo.name == "Error"
```

Comment:
I like this, much more solid than to search a string for XML

Reply:
But, RestXML supports error wrapping.
https://smithy.io/2.0/aws/protocols/aws-restxml-protocol.html#aws-protocols-restxml-trait

Will this code detect the error whether or not it is wrapped?

Reply:
In short, no if it is not Error at the root it won't detect it.

Reply:
OK.  Reading the SEP, it sounds pretty clear that the affected errors will be in the root.